### ISSUE TITLE
Consider Droping PHPCS 2.x support by August 31, 2018

### ISSUE THREAD

**Issue by photodude** [joomla/coding-standards#227]
Following what other Coding standards using PHPCS are doing, We should consider dropping PHPCS 2.x support by August 31, 2018

Example of what another Coding standard is doing (details of why are dropping support in August are included in their list)
https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1048

**Comment by photodude** [comment]
Note: The next release of PHPCS 2.9.x will be the last bug fix for that branch.

**Comment by wilsonge** [comment]
I think we'll leave the J3.x series on PHPCS 2.x series and J4.x will go straight onto PHPCS 3.x

**Comment by photodude** [comment]
Framework 2.x should also move onto PHPCS 3.x
Kinda always been what I suggested. If you don't support PHP 5.3 should plan for PHPCS 3.x

Of course, this is just a tool that we typically run on PHP 5.6+ so do we even care about the lack of PHP 5.3 support with the tool?

In any case, we need to get a PHPCS2.x stable releases and a PHPCS 3.x release out.

**Comment by mbabker** [comment]
> Framework 2.x should also move onto PHPCS 3.x

We'll get there.  I'm still working through dropping the gitmodule for 1.x and using Composer for 2.x across repos.  The ones that are already done is as easy as changing a version constraint, the rest still need some TLC.

**Comment by mbabker** [comment]
> Of course, this is just a tool that we typically run on PHP 5.6+ so do we even care about the lack of PHP 5.3 support with the tool?

Yes.  If someone clones a PHP 5.3 compat repo on a PHP 5.3 system they should be able to run all related test tooling and not rely on our CI systems to run tests for them.  It's part of the reason we're still going through the pain of keeping PHPUnit compatible with 4.8 even though there are now 3 newer major series.

**Comment by jrfnl** [comment]
Hiya, WordPress Coding Standards contributor here!

Just FYI as I noticed the WPCS ticket is referenced here:

[WPCS](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/) has had support for [PHPCS 2.x as well as 3.x](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1047) since August 2017, so we currently support both versions, though we encourage people to use PHPCS 3.x.

We did this as there are a number of company/other community standards within the WP community which depend on WPCS.
This way, those other standards will have a years time to upgrade their standard to be compatible with PHPCS 3.x
So when we drop support for PHPCS 2.x by end of August 2018, it will be after a year of cross-version PHPCS version support.

As a side-note - the [PHPCompatibility](https://github.com/wimg/PHPCompatibility) standard is compatible with PHPCS 1.5.6, 2.2+ and 3.0.2+ and intends to maintain PHPCS cross-version support.
